libtransmission/torrent-metainfo.cc

```diff
@@ -271,6 +271,11 @@ struct MetainfoHandler final : public transmission::benc::BasicHandler<MaxBencDe
             file_subpath_.clear();
             file_length_ = 0;
         }
+        if (pathStartsWith(InfoKey, FilesKey, ""sv, PathUtf8Key))
```

Comment:
Improvement: use `else if` here, since it's impossible to be simultaneously `pathIs(InfoKey, FilesKey)` (depth is 2) and `pathStartsWith(InfoKey, FilesKey, ""sv, PathUtf8Key)` (depth is 4 or more).

Reply:
Fixed in 69b285a75ca76018400faaa499065f1a30f0f942